addons/web_editor/static/src/components/media_dialog/media_dialog.js

```diff
@@ -167,11 +167,13 @@ export class MediaDialog extends Component {
                     if (style) {
                         element.setAttribute('style', style);
                     }
-                    if (this.props.media.dataset.shape) {
-                        element.dataset.shape = this.props.media.dataset.shape;
-                    }
-                    if (this.props.media.dataset.shapeColors) {
-                        element.dataset.shapeColors = this.props.media.dataset.shapeColors;
+                    if (this.state.activeTab === TABS.IMAGES.id) {
+                        if (this.props.media.dataset.shape) {
+                            element.dataset.shape = this.props.media.dataset.shape;
+                        }
+                        if (this.props.media.dataset.shapeColors) {
+                            element.dataset.shapeColors = this.props.media.dataset.shapeColors;
+                        }
```

Comment:
While it works well in the cases images ->  {images, icons, videos, documents} there is one remaining case that should be taken into account: images -> illustrations (so the non-unsplash images when you type something in the searchbar)
Indeed, these images cannot have a shape (they don't have the option) but when replacing an image with a shape by an illustration, it keeps the shape data-attributes (because we are in the "Images" tab) and so the shape option appears, but not correctly (see image)
![image](https://github.com/odoo/odoo/assets/102796037/a44ed8e8-d11e-4eb9-9cd5-b739d6dd0cf0)
And if you try to change the color (since the option is there), then there is a traceback :grimacing:

So maybe you should add something to support this case ? :slightly_smiling_face: